services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedTenantService.java

```diff
@@ -460,11 +460,12 @@ public void update(final String tenantId, final JsonObject tenantSpec, final Opt
         }
     }
 
+    @SuppressWarnings("unchecked")
     private Tenant convertTenantObject(final TenantObject tenantObject) {
         return new Tenant()
                 .setEnabled(tenantObject.isEnabled())
                 .setExtensions((JsonObject) tenantObject.getProperty("ext"))
-                .setAdapters(tenantObject.getAdapterConfigurations())
+                .setAdapters(tenantObject.getAdapterConfigurations().getList())
```

Comment:
but if no adapter is specified you get NPE :) That why I added a second method in the first place